source/application/IDEAS/IDEASAppController.cpp

```diff
@@ -356,48 +319,37 @@ void IDEASAppController::configureSingleSpectrumView(AMGenericScanEditor *editor
 
 void IDEASAppController::onUserConfigurationLoadedFromDb()
 {
-	QList<IDEASScanConfiguration *> configurations = QList<IDEASScanConfiguration *>()
-			<< xasScanConfiguration_
-			<< mapScanConfiguration_;
-
-	foreach (IDEASScanConfiguration *configuration, configurations)
-		configuration->setFluorescenceDetector(ideasUserConfiguration_->fluorescenceDetector());
-
-	AMXRFDetector *detector = IDEASBeamline::ideas()->ketek();
-
-	foreach (AMRegionOfInterest *region, ideasUserConfiguration_->regionsOfInterest()){
-
-		detector->addRegionOfInterest(region);
-		mapScanConfiguration_->addRegionOfInterest(region);
-		xasScanConfiguration_->addRegionOfInterest(region);
-		genericConfiguration_->addRegionOfInterest(region);
+	CLSAppController::onUserConfigurationLoadedFromDb();
+
+	IDEASUserConfiguration *ideasUserConfiguration = qobject_cast<IDEASUserConfiguration *>(userConfiguration_);
+	if (ideasUserConfiguration) {
+		IDEAS::FluorescenceDetectors fluorescenceDetector = ideasUserConfiguration->fluorescenceDetector();
+		if (xasScanConfiguration_)
+			xasScanConfiguration_->setFluorescenceDetector(fluorescenceDetector);
+		if (mapScanConfiguration_)
+			mapScanConfiguration_->setFluorescenceDetector(fluorescenceDetector);
 	}
-
-	// This is connected here because we want to listen to the detectors for updates, but don't want to double add regions on startup.
-	connect(IDEASBeamline::ideas()->ketek(), SIGNAL(addedRegionOfInterest(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestAdded(AMRegionOfInterest*)));
-	connect(IDEASBeamline::ideas()->ketek(), SIGNAL(removedRegionOfInterest(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestRemoved(AMRegionOfInterest*)));
-	connect(IDEASBeamline::ideas()->ketek(), SIGNAL(regionOfInterestBoundingRangeChanged(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestBoundingRangeChanged(AMRegionOfInterest*)));
 }
 
 void IDEASAppController::onRegionOfInterestAdded(AMRegionOfInterest *region)
 {
-	ideasUserConfiguration_->addRegionOfInterest(region);
+	userConfiguration_->addRegionOfInterest(region);
 	mapScanConfiguration_->addRegionOfInterest(region);
```

Comment:
The reason I didn't update `userConfiguration_` in the CLS-level is that there is no generalized way to update the scan configuration in CLS-level at this moment.

Maybe in the future we will do that